packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/SharedPreferencesNotificationsStore.kt

```diff
@@ -0,0 +1,167 @@
+   */
+  @Throws(IOException::class)
+  fun saveNotificationRequest(notificationRequest: NotificationRequest) =
+    sharedPreferences.edit()
+      .putString(
+        preferencesNotificationRequestKey(notificationRequest.identifier),
+        serializeNotificationRequest(notificationRequest)
+      )
+      .apply()
+
+
+  /**
+   * Removes notification info for given identifier.
+   *
+   * @param identifier Notification identifier
+   */
+  fun removeNotificationRequest(identifier: String) =
+    removeNotificationRequest(sharedPreferences.edit(), identifier).apply()
+
+  /**
+   * Perform notification removal on provided [SharedPreferences.Editor] instance. Can be reused
+   * to batch deletion.
+   *
+   * @param editor     Editor to apply changes onto
+   * @param identifier Notification identifier
+   * @return Returns a reference to the same Editor object, so you can
+   * chain put calls together.
+   */
+  private fun removeNotificationRequest(editor: SharedPreferences.Editor, identifier: String) =
+    editor.remove(preferencesNotificationRequestKey(identifier))
+
+  /**
+   * Removes all notification infos, returning removed IDs.
+   */
+  fun removeAllNotificationRequests(): Collection<String> =
+    with(sharedPreferences.edit()) {
+      allNotificationRequests.map {
+        removeNotificationRequest(this, it.identifier)
+        it.identifier
+      }.let {
+        this.apply()
+        it
+      }
+    }
+
+  /**
+   * Serializes the trigger to a Base64-encoded string.
+   *
+   * @param notificationRequest Notification request to serialize
+   * @return Base64-encoded, serialized trigger
+   * @throws IOException Thrown if there is an error while writing trigger to string.
+   */
+  @Throws(IOException::class)
+  private fun serializeNotificationRequest(notificationRequest: NotificationRequest) =
+    ByteArrayOutputStream().use { byteArrayOutputStream ->
+      ObjectOutputStream(byteArrayOutputStream).use { objectOutputStream ->
+        objectOutputStream.writeObject(notificationRequest)
+        Base64.encodeToString(byteArrayOutputStream.toByteArray(), Base64.NO_WRAP)
+      }
+    }
+
+  /**
+   * Deserializes trigger from the string representation.
+   *
+   * @param trigger Base64-encoded, serialized trigger representation
+   * @return Deserialized trigger
+   * @throws IOException            Thrown if there is an error while reading trigger from String
+   * @throws ClassNotFoundException Thrown if the deserialization failes due to class not being found.
+   * @throws InvalidClassException  Thrown if the trigger is of invalid class.
+   */
+  @Throws(IOException::class, ClassNotFoundException::class, InvalidClassException::class)
+  private fun deserializeNotificationRequest(trigger: String?): NotificationRequest =
+    Base64.decode(trigger, Base64.NO_WRAP).let {
+      ByteArrayInputStream(it).use { byteArrayInputStream ->
+        ObjectInputStream(byteArrayInputStream).use { ois ->
```

Comment:
It seems similar to the deserialisation from the other file? 🤔 
Maybe it's ok to ship some helper method for this?

Reply:
I'm lovin it! https://github.com/expo/expo/pull/10765/commits/d5d0b24a608e4f04fd0689882e3d0f9713a9ee72